src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -391,6 +391,17 @@ def is_equal_cell(self, other) -> bool:
             return False
         return True
 
+    def get_chemical_symbols(self) -> list:
+        """
+        Get the chemical symbols of the atoms in the atomic cell. These are defined on `atoms_state[*].chemical_symbol`.
+
+        Returns:
+            list: The list of chemical symbols of the atoms in the atomic cell.
+        """
+        if not self.atoms_state:
+            return []
+        return [atom_state.chemical_symbol for atom_state in self.atoms_state]
```

Comment:
Wouldn't this raise an `AttributeError` if `atom_state.chemical_symbol` is not set? It could also result in `None`, which should be filtered out too.

Overall, I'm also quite a fan list comprehensions. When dealing with edge cases in Python, however, they often fall short... There are languages that do a better job at combining both.